controller_test.go

```diff
@@ -0,0 +1,61 @@
+package main
+
+import (
+	"context"
+	"fmt"
+	"testing"
+
+	compute "cloud.google.com/go/compute/apiv1"
+	"cloud.google.com/go/compute/apiv1/computepb"
+	"github.com/googleapis/gax-go/v2"
+	"k8s.io/apimachinery/pkg/types"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+	"sigs.k8s.io/controller-runtime/pkg/reconcile"
+)
+
+type runtimeClientImpt struct {
+	labels map[string]string
+}
+
+func (r *runtimeClientImpt) Get(ctx context.Context, key types.NamespacedName, obj client.Object, opts ...client.GetOption) error {
+	fmt.Println("Called Get")
+	obj.SetLabels(r.labels)
+	return nil
+}
+
+type instanceClientImpl struct {
+	calledWithSecurityPolicy string
+}
+
+func (i *instanceClientImpl) SetSecurityPolicy(ctx context.Context, req *computepb.SetSecurityPolicyInstanceRequest, opts ...gax.CallOption) (*compute.Operation, error) {
+	fmt.Println("Called SetSecurityPolicy")
+	i.calledWithSecurityPolicy = *req.InstancesSetSecurityPolicyRequestResource.SecurityPolicy
```

Comment:
add a counter to ensure it's only called once?

Reply:
Done, added the other fields in the SetSecurityPolicyInstanceRequest to check as well